Various readme edits

lonewsk opened the pull request.

lonewsk committed:
Readme edits

lonewsk committed:
Readme edits

lonewsk committed:
Readme edits

yatli commented on README.md:
````diff
@@ -325,7 +323,7 @@ $ exit
 ### Unmount The Root Filesystem
 
 ```
- # umonut root
````
I saw this last time but the edit was `umonut -> unmount` 🤣

lonewsk replied:
I'm gonna stop trying to correct it ahah

yatli commented on README.md:
````diff
@@ -358,63 +356,64 @@ but use our tarball in place of theirs.
 1. Zero the beginning of the SD card
 
 ```
-# dd if=/dev/zero of=/dev/sdX bs=1M count=32
````
this one. it depends on how the uSD is connected. I was doing it with a card reader so it showed up as /dev/sdb

lonewsk replied:
I thought mmcblk would hint in the right direction and prevent wrong targets. But both are okay.

yatli commented on README.md:
```diff
@@ -358,63 +356,64 @@ but use our tarball in place of theirs.
-    6. Write the partition table and exit by typing **w**
+    3. Type **n**, then **p** for primary, **1** for the first partition on the drive, **32768** for the first sector, then **+2G** for the last sector (boot partition, 2GB)
+	4. Type **p** and note the end sector number for the partition freshly created.
+    5. Type **n**, then **p**, **2** for the second partition, **use number from step 4 and add 2** as first sector, then **+4G** for the last sector (swap partition, 4GB)
```
I created these instructions without actually running it.. Purely imaginative. :D
Can you explain the "add 2" part here?

lonewsk replied:
Since the first partition created (boot) isn't starting at sector 0 (instead 32768), the next partition will not start *after* the first one, but on sector 0. That's why you need to retrieve the previous partition's end sector, and add at least 1. I decided to burn a sector for safety, thus 2.

yatli replied:
aah, correct! my imagined machine in my head totally ignored that.

yatli commented on README.md:
````diff
@@ -358,63 +356,64 @@ but use our tarball in place of theirs.
+# mkfs.f2fs -l ROOT_ARCH -O extra_attr,inode_checksum,sb_checksum /dev/mmcblkX3
 ```
 
+**IMPORTANT** The `mkswap` command will return the swap partition's UUID, which will be needed later. Please write it down.
````
btw if we can make sure fdisk gives us `gpt` partitions, then `systemd` will automatically find and use the swap partitions.

lonewsk replied:
oh nice, that could be handy!

yatli replied:
see: https://wiki.archlinux.org/title/swap#Activation_by_systemd

lonewsk committed:
readme edits

yatli commented on README.md:
```diff
@@ -432,8 +431,7 @@ The SD card is now ready to be booted by the DevTerm! Good luck!
 
 ## Next Steps
 
-You will want to set up Wi-Fi on first boot. You can do so by
```
could you also test if wiringpi and the printer works?
also, while I'm not sure if it works, you can test suspend without the mainboard mod like this:
```
1. sudo systemctl suspend
2. observe that it goes to sleep
3. plug in power
4. observe whether it resumes
5. if we're lucky and it bounces back, use dmesg to verify it's a deep sleep
```

lonewsk replied:
Should the green led be on while sleeping? 

After running systemctl suspend, screen turns off, keyboard unresponsive, green light on.

It immediately resume when I plug a power cable in.

lonewsk replied:
I don't have the printer to try it sorry.

lonewsk replied:
And fan works 👍

yatli replied:
ooh then it means we actually have a way to wake up the core without the mainboard hack. this is good.
no worry about the printer.

yatli replied:
> And fan works

yep that's the handy work of the Manjaro folks.

yatli commented:
I'm actually (happily) a little bit surprised that it works, given that my image is like half the size of cuu's -- I suspected that I removed important bits in the cleanup step.

I extracted `/etc` of both and compared them, nothing important is missing.

lonewsk commented:
> I'm actually (happily) a little bit surprised that it works, given that my image is like half the size of cuu's -- I suspected that I removed important bits in the cleanup step.
> 
> I extracted `/etc` of both and compared them, nothing important is missing.

That's Arch's magic right there, I guess. Plus it's really stripped down compared to armbian. I'm not as much surprised by the image's size.

yatli commented:
I mean cuu's original ArchLinuxARM rootfs, not guu's Armbian :D :D
btw for a while I thought they're both on cpi team, and cpi team somewhat assign aliases to their employees in a `[a-z][A-Z][0-9]uu` manner

yatli commented:
take your time, and let me know when you feel it's ready.
🛌 now

lonewsk commented:
Looks okay to me! I think you can merge :)

yatli commented:
Go.

yatli commented:
Thanks a lot!

lonewsk commented:
> Thanks a lot!

You're the one that made all the work ;)
